pkg/graveler/settings/manager.go

```diff
@@ -105,52 +108,52 @@ func computeChecksum(value []byte) (*string, error) {
 	return swag.String(hex.EncodeToString(h.Sum(nil))), nil
 }
 
-// GetLatest returns the latest setting under the given repository and key, without using the cache.
-// The returned checksum represents the version of the setting, and can be passed to SaveIf for conditional updates.
-func (m *Manager) GetLatest(ctx context.Context, repository *graveler.RepositoryRecord, key string, settingTemplate proto.Message) (proto.Message, *string, error) {
+// GetLatest loads the latest setting into dst.
+// It returns a checksum representing the version of the setting, which can be passed to SaveIf for conditional updates.
+// The checksum of a non-existent setting is the empty string.
+func (m *Manager) GetLatest(ctx context.Context, repository *graveler.RepositoryRecord, key string, dst proto.Message) (*string, error) {
 	settings, err := m.store.Get(ctx, []byte(graveler.RepoPartition(repository)), []byte(graveler.SettingsPath(key)))
 	if err != nil {
 		if errors.Is(err, kv.ErrNotFound) {
 			err = graveler.ErrNotFound
 		}
-		return nil, nil, err
+		return swag.String(""), err
```

Comment:
I understood why we would like to return a value in case of an error, specific not found. But I hope that the caller will work based on the error and the convention of the API and not use the checksum value in case of an error.

Reply:
Done - not returning empty string.